### PR TITLE
Improve formatting for balances in FarmCards

### PR DESCRIPTION
**Pull request by KiChjang**
Attempt at fixing https://www.notion.so/Kanban-Board-c5a1f2345d3c430ca73d9377564b2073?p=4df67c4e78194c27862fcc6ae05a1b3c

### CODE REVIEW

**Comment by KiChjang** [comment]
@belbix Could I get a review on this?

File: src/components/farmCards/FarmCard.jsx [hunk 1]
```diff
@@ -17,8 +17,13 @@ export default function FarmCard({ summary_information }) {
         return pool.address === summary_information.address;
       });
     }
+  };
 
-  }
+  const formatBalance = (balance) => {
+    const bal = parseFloat(balance);
+    if (bal === 0) return bal;
+    return bal < 1 ? bal.toFixed(6) : bal.toFixed(1);
```
Comment (belbix) [inline src/components/farmCards/FarmCard.jsx:25]: please, develop a more complex logic
ideally if we will have the same width for all reasonable figures

File: src/components/farmCards/FarmCard.jsx [hunk 2]
```diff
@@ -17,8 +17,19 @@ export default function FarmCard({ summary_information }) {
...
+  const formatBalance = (balance) => {
+    const bal = parseFloat(balance);
+    if (bal === 0) return bal;
+    if (bal < 10) return bal.toFixed(6);
+    if (bal < 100) return bal.toFixed(5);
+    if (bal < 1000) return bal.toFixed(4);
+    if (bal < 10000) return bal.toFixed(3);
+    if (bal < 100000) return bal.toFixed(2);
+    if (bal < 1000000) return bal.toFixed(1);
+    return Math.round(bal);
```
Comment (KiChjang) [inline src/components/farmCards/FarmCard.jsx:31]: @belbix How does this look?

Comment (belbix) [reply src/components/farmCards/FarmCard.jsx:31]: after discussion with team members, we decided to make it in another way
please keep 6 figures
but create a function that prettifies numbers by the following logic:
10000 - 10k
10000000 - 10m
...
+ create a tooltip with the full amount

**Comment by KiChjang** [comment]
@belbix Here's how it looks like with my latest commit:
image

File: src/components/farmCards/FarmCard.jsx [hunk 3]
```diff
@@ -17,8 +18,23 @@ export default function FarmCard({ summary_information }) {
...
+    if (bal < 100) return bal.toFixed(5).toString();
+    if (bal < 1000) return bal.toFixed(4).toString();
+    if (bal < 10_000) return bal.toFixed(3).toString();
+    if (bal < 100_000) return (bal / 1000).toFixed(5).toString() + "k";
+    if (bal < 1_000_000) return (bal / 1000).toFixed(4).toString() + "k";
+    if (bal < 10_000_000) return (bal / 1000).toFixed(3).toString() + "k";
+    if (bal < 100_000_000) return (bal / 1_000_000).toFixed(5).toString() + "m";
+    if (bal < 1_000_000_000) return (bal / 1_000_000).toFixed(4).toString() + "m";
+    if (bal < 10_000_000_000) return (bal / 1_000_000).toFixed(3).toString() + "m";
+    return Math.round(bal).toString();
```
Comment (cryptopower80) [inline src/components/farmCards/FarmCard.jsx:36]: I think we can make it easier if we just keep two digits and use this library https://www.npmjs.com/package/js-abbreviation-number

**Comment by cryptopower80** [comment]
> @belbix Here's how it looks like with my latest commit:
> image

Screenshot 2021-03-05 at 04 07 33

Would it be possible to use this style?

**Comment by yiepier** [comment]
@KiChjang how is it coming with this pr?

**Comment by KiChjang** [comment]
@yiepier I'm not sure who I should listen to. There are quite a few conflicting opinions on how the UI should look like, and I'm unsure how to proceed.

**Comment by yiepier** [comment]
I can't see that info in this pr, can you tell me whose opinions conflict?

**Comment by KiChjang** [comment]
@yiepier @belbix has told me in [this comment](https://github.com/harvestfi/dashboard/pull/78#discussion_r587745178) that he has consulted with the community that it should always consistently display 6 figures, but @cryptopower80 [reviewed](https://github.com/harvestfi/dashboard/pull/78#discussion_r587998203) later on and suggested to use a library that keeps only 2 digits.

File: src/components/farmCards/FarmCard.jsx (hunk 3 above)
Comment (belbix) [reply src/components/farmCards/FarmCard.jsx:36]: two digits not enough for some assets, like BTC

**Comment by belbix** [comment]
use the lib and keep 6 digits
we can easily adjust digits quantity if it will be required
btw better write me in DM on Discord, I have too many updates from GitHub

**Comment by yiepier** [comment]
Closing this pr because of no response and too many conflicts.
you can reopen it if you want, @KiChjang

**Pull request closed without merging**